File: src/main/kotlin/magicbook/gtlitecore/common/entity/explosive/EntityNaquadriaCharge.java
package magicbook.gtlitecore.common.entity.explosive;

import gregtech.common.entities.EntityGTExplosive;
import magicbook.gtlitecore.common.block.GTLiteMetaBlocks;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.world.World;
import org.jetbrains.annotations.NotNull;

public class EntityNaquadriaCharge extends EntityGTExplosive
{

    @SuppressWarnings("unused")
    public EntityNaquadriaCharge(World world)
    {
        super(world);
    }

    public EntityNaquadriaCharge(World world, double x, double y, double z, EntityLivingBase exploder)
    {
        super(world, x, y, z, exploder);
    }

    @Override
    protected float getStrength()
    {
        return 200.0F;
    }

    @Override
    public boolean dropsAllBlocks()
    {
        return false;
    }

    @Override
    protected int getRange()
    {
        return 64;
    }

    @NotNull
    @Override
    public IBlockState getExplosiveState()
    {
        return GTLiteMetaBlocks.NAQUADRIA_CHARGE.getDefaultState();
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/common/entity/explosive/EntityNaquadriaCharge.kt
package gregtechlite.gtlitecore.common.entity.explosive

import gregtech.common.entities.EntityGTExplosive
import gregtechlite.gtlitecore.common.block.GTLiteMetaBlocks
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.EntityLivingBase
import net.minecraft.world.World

class EntityNaquadriaCharge(world: World?,
                            x: Double, y: Double, z: Double,
                            exploder: EntityLivingBase?) : EntityGTExplosive(world, x, y, z, exploder)
{

    constructor(world: World?) : this(world, 0.0, 0.0, 0.0, null)

    override fun getStrength() = 200.0f

    override fun dropsAllBlocks() = false

    override fun getRange() = 64

    override fun getExplosiveState(): IBlockState = GTLiteMetaBlocks.NAQUADRIA_CHARGE.defaultState

}

